parallel-consumer-core/src/main/java/io/confluent/parallelconsumer/internal/AbstractParallelEoSStreamProcessor.java

```diff
@@ -614,7 +614,7 @@ private void addInstanceMDC() {
         log.trace("Loop: Process mailbox");
```

Comment:
GH won't let me comment on the correct line, but 

Shouldn't this only happen when running? and NOT paused?

`int newWork = handleWork(userFunction, callback);`

Reply:
There is a state check inside `handleWork`. Adding another if here would probably only make sense when also refactoring `handleWork` internals (e.g. pulling `checkPressure()` out of handleWork).

Reply:
As discussed via chat - refactoring of `handleWork` won't be done as part of this PR.